### PR TITLE
Add new skill for custom markdown to html covnersion

### PR DESCRIPTION
**Pull request by jhauga**
## Pull Request Checklist

- [x] I have read and followed the [CONTRIBUTING.md](https://github.com/github/awesome-copilot/blob/main/CONTRIBUTING.md) guidelines.
- [x] My contribution adds a new instruction, prompt, agent, or skill file in the correct directory.
- [x] The file follows the required naming convention.
- [x] The content is clearly structured and follows the example format.
- [x] I have tested my instructions, prompt, agent, or skill with GitHub Copilot.
- [x] I have run `npm start` and verified that `README.md` is up to date.

---

### Testing

Did a control branch, and a test branch; which used the skill.

#### Disclaimer

The test branch produced near production ready tool, but to be honest - I wasn't sure about making
this PR, as the control test did better on initial prompt, but since there are several contexts
for which the skill can be used, I thought I would make the PR anyway as I'm still going to use the skill
locally for some tools, and would like to be able to download to different projects via
[vscode-awesome-copilot](https://github.com/timheuer/vscode-awesome-copilot).

### [Control Branch](https://github.com/jhauga/support-repo/tree/negative-markdown-to-html)

Only using the reference files with the prompt file [markdown-to-html.prompt.md](https://github.com/jhauga/support-repo/tree/negative-markdown-to-html/.github/prompts/markdown-to-html.prompt.md).

### [Test Branch](https://github.com/jhauga/support-repo/tree/markdown-to-html)

I prompted a bit more in favor of the test using the skill, polishing the results with over 10
prompts total. But the results speak for themselves. The branch that used the skill is debatable
a ready to publish tool for converting markdown to html with an additional feature of easily
applying html attributes to the parsed html data.

---

## Type of Contribution

- [ ] New instruction file.
- [ ] New prompt file.
- [ ] New agent file.
- [ ] New collection file.
- [x] New skill file.
- [ ] Update to existing instruction, prompt, agent, collection or skill.
- [ ] Other (please specify):

---

By submitting this pull request, I confirm that my contribution abides by the [Code of Conduct](../CODE_OF_CONDUCT.md) and will be licensed under the MIT License.

### CODE REVIEW

File: skills/markdown-to-html/references/basic-markdown.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+```
+
+![Screenshot of rendered GitHub Markdown showing sample h1, h2, and h3 headers, which descend in type size and visual weight to show hierarchy level.](https://docs.github.com/assets/images/help/writing/headings-rendered.png)
+
+When you use two or more headings, GitHub automatically generates a table of contents that you can access by clicking the "Outline" menu icon <svg version="1.1" width="16" height="16" viewBox="0 0 16 16" class="octicon octicon-list-unordered" aria-label="Table of Contents" role="img"><path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"></path></svg> within the file header. Each heading title is listed in the table of contents and you can click a title to navigate to the selected section.
+
+![Screenshot of a README file with the drop-down menu for the table of contents exposed. The table of contents icon is outlined in dark orange.](https://docs.github.comhttps://docs.github.com/assets/images/help/repository/headings-toc.png)
```
Comment (jhauga) [inline skills/markdown-to-html/references/basic-markdown.md]:
```suggestion
![Screenshot of a README file with the drop-down menu for the table of contents exposed. The table of contents icon is outlined in dark orange.](https://docs.github.com/assets/images/help/repository/headings-toc.png)
```

File: skills/markdown-to-html/references/basic-markdown.md [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+
+```markdown
+Text that is not a quote
+
+> Text that is a quote
+```
+
+Quoted text is indented with a vertical line on the left and displayed using gray type.
+
+![Screenshot of rendered GitHub Markdown showing the difference between normal and quoted text.](https://docs.github.comhttps://docs.github.com/assets/images/help/writing/quoted-text-rendered.png)
```
Comment (jhauga) [inline skills/markdown-to-html/references/basic-markdown.md]:
```suggestion
![Screenshot of rendered GitHub Markdown showing the difference between normal and quoted text.](https://docs.github.com/assets/images/help/writing/quoted-text-rendered.png)
```

File: skills/markdown-to-html/references/basic-markdown.md [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+This example
+
+Will have a blank line separating both lines
+```
+
+## Images
+
+You can display an image by adding <kbd>!</kbd> and wrapping the alt text in `[ ]`. Alt text is a short text equivalent of the information in the image. Then, wrap the link for the image in parentheses `()`.
+
+`![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://docs.github.comhttps://myoctocat.com/assets/images/base-octocat.svg)`
```
Comment (jhauga) [inline skills/markdown-to-html/references/basic-markdown.md]:
```suggestion
`![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://myoctocat.com/assets/images/base-octocat.svg)`
```

File: skills/markdown-to-html/references/code-blocks.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+````
+```
+Look! You can see my backticks.
+```
+````
+`````
+
+![Screenshot of rendered Markdown showing that when you write triple backticks between quadruple backticks they are visible in the rendered content.](https://docs.github.com/assets/images/help/writing/fenced-code-show-backticks-rendered.png)
+
+If you are frequently editing code snippets and tables, you may benefit from enabling a fixed-width font in all comment fields on GitHub. For more information, see [About writing and formatting on GitHub](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github#enabling-fixed-width-fonts-in-the-editor).
```
Comment (jhauga) [inline skills/markdown-to-html/references/code-blocks.md]:
```suggestion
If you are frequently editing code snippets and tables, you may benefit from enabling a fixed-width font in all comment fields on GitHub. For more information, see [About writing and formatting on GitHub](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github#enabling-fixed-width-fonts-in-the-editor).
```

File: skills/markdown-to-html/references/code-blocks.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+puts markdown.to_html
+```
+````
+
+This will display the code block with syntax highlighting:
+
+![Screenshot of three lines of Ruby code as displayed on GitHub. Elements of the code display in purple, blue, and red type for scannability.](https://docs.github.com/assets/images/help/writing/code-block-syntax-highlighting-rendered.png)
+
+> \[!TIP]
+> When you create a fenced code block that you also want to have syntax highlighting on a GitHub Pages site, use lower-case language identifiers. For more information, see [About GitHub Pages and Jekyll](https://docs.github.comn/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#syntax-highlighting).
```
Comment (jhauga) [inline skills/markdown-to-html/references/code-blocks.md]:
```suggestion
> When you create a fenced code block that you also want to have syntax highlighting on a GitHub Pages site, use lower-case language identifiers. For more information, see [About GitHub Pages and Jekyll](https://docs.github.com/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#syntax-highlighting).
```

File: skills/markdown-to-html/references/code-blocks.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+We use [Linguist](https://github.com/github-linguist/linguist) to perform language detection and to select [third-party grammars](https://github.com/github-linguist/linguist/blob/main/vendor/README.md) for syntax highlighting. You can find out which keywords are valid in [the languages YAML file](https://github.com/github-linguist/linguist/blob/main/lib/linguist/languages.yml).
+
+## Creating diagrams
+
+You can also use code blocks to create diagrams in Markdown. GitHub supports Mermaid, GeoJSON, TopoJSON, and ASCII STL syntax. For more information, see [Creating diagrams](https://docs.github.comn/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams).
```
Comment (jhauga) [inline skills/markdown-to-html/references/code-blocks.md]:
```suggestion
You can also use code blocks to create diagrams in Markdown. GitHub supports Mermaid, GeoJSON, TopoJSON, and ASCII STL syntax. For more information, see [Creating diagrams](https://docs.github.com/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams).
```

File: skills/markdown-to-html/references/code-blocks.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## Further reading
+
+* [GitHub Flavored Markdown Spec](https://github.github.com/gfm/)
+* [Basic writing and formatting syntax](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
```
Comment (jhauga) [inline skills/markdown-to-html/references/code-blocks.md]:
```suggestion
* [Basic writing and formatting syntax](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
```

File: skills/markdown-to-html/references/tables.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+```markdown
+| Command | Description |
+| --- | --- |
+| git status | List all new or modified files |
+| git diff | Show file differences that haven't been staged |
+```
+
+![Screenshot of a GitHub Markdown table with two columns of differing width. Rows list the commands "git status" and "git diff" and their descriptions.](https://docs.github.com/assets/images/help/writing/table-varied-columns-rendered.png)
+
+If you are frequently editing code snippets and tables, you may benefit from enabling a fixed-width font in all comment fields on GitHub. For more information, see [About writing and formatting on GitHub](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github#enabling-fixed-width-fonts-in-the-editor).
```
Comment (jhauga) [inline skills/markdown-to-html/references/tables.md]:
```suggestion
If you are frequently editing code snippets and tables, you may benefit from enabling a fixed-width font in all comment fields on GitHub. For more information, see [About writing and formatting on GitHub](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github#enabling-fixed-width-fonts-in-the-editor).
```

File: skills/markdown-to-html/references/tables.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+## Formatting content within your table
+
+You can use [formatting](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) such as links, inline code blocks, and text styling within your table:
```
Comment (jhauga) [inline skills/markdown-to-html/references/tables.md]:
```suggestion
You can use [formatting](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) such as links, inline code blocks, and text styling within your table:
```

File: skills/markdown-to-html/references/tables.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+| Backtick | `         |
+| Pipe     | \|        |
+```
+
+![Screenshot of a Markdown table as rendered on GitHub showing how pipes, which normally close cells, are shown when prefaced by a backslash.](https://docs.github.com/assets/images/help/writing/table-escaped-character-rendered.png)
+
+## Further reading
+
+* [GitHub Flavored Markdown Spec](https://github.github.com/gfm/)
+* [Basic writing and formatting syntax](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
```
Comment (jhauga) [inline skills/markdown-to-html/references/tables.md]:
```suggestion
* [Basic writing and formatting syntax](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
```

File: skills/markdown-to-html/references/writing-mathematical-expressions.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+GitHub's math rendering capability uses MathJax; an open source, JavaScript-based display engine. MathJax supports a wide range of LaTeX macros, and several useful accessibility extensions. For more information, see [the MathJax documentation](http://docs.mathjax.org/en/latest/input/tex/index.html#tex-and-latex-support) and [the MathJax Accessibility Extensions Documentation](https://mathjax.github.io/MathJax-a11y/docs/#reader-guide).
+
+Mathematical expressions rendering is available in GitHub Issues, GitHub Discussions, pull requests, wikis, and Markdown files.
+
+## Writing inline expressions
+
+There are two options for delimiting a math expression inline with your text. You can either surround the expression with dollar symbols (`$`), or start the expression with <code>$\`</code> and end it with <code>\`$</code>. The latter syntax is useful when the expression you are writing contains characters that overlap with markdown syntax. For more information, see [Basic writing and formatting syntax](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).
```
Comment (jhauga) [inline skills/markdown-to-html/references/writing-mathematical-expressions.md]:
```suggestion
There are two options for delimiting a math expression inline with your text. You can either surround the expression with dollar symbols (`$`), or start the expression with <code>$\`</code> and end it with <code>\`$</code>. The latter syntax is useful when the expression you are writing contains characters that overlap with markdown syntax. For more information, see [Basic writing and formatting syntax](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).
```

File: skills/markdown-to-html/references/writing-mathematical-expressions.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+This sentence uses $\` and \`$ delimiters to show math inline: $`\sqrt{3x-1}+(1+x)^2`$
+```
+
+![Screenshot of rendered Markdown showing an inline mathematical expression with backtick syntax: the square root of 3x minus 1 plus (1 plus x) squared.](https://docs.github.com/assets/images/help/writing/inline-backtick-math-markdown-rendering.png)
+
+## Writing expressions as blocks
+
+To add a math expression as a block, start a new line and delimit the expression with two dollar symbols `$$`.
+
+>  [!TIP] If you're writing in an .md file, you will need to use specific formatting to create a line break, such as ending the line with a backslash as shown in the example below. For more information on line breaks in Markdown, see [Basic writing and formatting syntax](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#line-breaks).
```
Comment (jhauga) [inline skills/markdown-to-html/references/writing-mathematical-expressions.md]:
```suggestion
>  [!TIP] If you're writing in an .md file, you will need to use specific formatting to create a line break, such as ending the line with a backslash as shown in the example below. For more information on line breaks in Markdown, see [Basic writing and formatting syntax](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#line-breaks).
```

File: skills/markdown-to-html/references/writing-mathematical-expressions.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+  ```text
+  To split <span>$</span>100 in half, we calculate $100/2$
+  ```
+
+  ![Screenshot of rendered Markdown showing how span tags around a dollar sign display the sign as inline text not as part of a mathematical equation.](https://docs.github.com/assets/images/help/writing/dollar-sign-inline-math-expression.png)
+
+## Further reading
+
+* [The MathJax website](http://mathjax.org)
+* [Getting started with writing and formatting on GitHub](https://docs.github.comn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github)
```
Comment (jhauga) [inline skills/markdown-to-html/references/writing-mathematical-expressions.md]:
```suggestion
* [Getting started with writing and formatting on GitHub](https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github)
```
